File: packages/backend/src/main/java/com/bookspank/backend/service/BookService.java
package com.bookspank.backend.service;

import java.util.List;
import java.util.Optional;

import com.bookspank.backend.dto.PostBookForm;
import com.bookspank.backend.model.Book;

public interface BookService {
    public List<Book> getBooks(Integer limit, Integer offset);

    public List<Book> getCompletedBooks(Integer limit, Integer offset);

    public List<Book> getUserBooks(Integer userId, Integer limit, Integer offset);

    public void postUserBook(Integer userId, PostBookForm form);

    public Optional<Book> getCurrentBook();

}

File: packages/aws-nextjs/lib/service/BookService.ts
import { Book } from "@/model/Book";
import { OpenLibrarySearchResponse } from "../dto/OpenLibrarySearchResponse";

export interface IBookService {

    getCompleted(limit: number, offset: number): Promise<Book[]>;

    getUserBooks(userId: number, limit: number, offset: number): Promise<Book[]>;

    addBook(userId: number, bookInfo: OpenLibrarySearchResponse): Promise<void>;

    getCurrentBook(): Promise<Book | null>;
}